woocommerce/woocommerce-android
# Replace Fabric with Sentry

Adds support for using Sentry via Tracks.

Most of the logic is over in https://github.com/Automattic/Automattic-Tracks-Android/pull/46. Once that's merged, I'll update this PR with a real version number, instead of using a Jitpack commit hash reference.

**To Test**
1. Add the ability to manually trigger a crash (putting `CrashLogging.crash()` somewhere will do that nicely).
2. Create a production version of the app using `./gradlew assembleVanillaRelease`.
3. Copy the `.apk` file to your device.
4. Log in, ensuring that your user is set up to send crash reports.
5. Trigger the crash
6. Re-run the app, and let it idle for a moment (this'll ensure the crash report goes to Sentry)
7. Note your shiny new crash entry in Sentry (it'll likely get attached to an existing sample crash from my testing). However yours should be the most recent event.
